iroh-bytes/src/get.rs

```diff
@@ -78,6 +78,21 @@ pub async fn run_ticket(
     )
     .await?;
 
+    let request = if ticket.token().is_some() && request.token().is_none() {
+        // we have a ticket, but no token, so we need to add the token to the request
+        match request {
+            AnyGetRequest::Get(get_request) => {
+                AnyGetRequest::Get(get_request.with_token(ticket.token().cloned()))
+            }
+            AnyGetRequest::CustomGet(mut custom_get_request) => {
+                custom_get_request.token = ticket.token().cloned();
+                AnyGetRequest::CustomGet(custom_get_request)
+            }
+        }
+    } else {
+        request
+    };
```

Comment:
This is kinda terrible, but I think the right approach.

Reply:
fine for now, we just really need to get rid of the `Ticket` in the internal handling, there is just too much duplication between options, but that's for another PR

Reply:
filed #1159